### DISCUSSION TITLE
Unable to configure TX_PWR in project config when building

### DISCUSSION THREAD

**Discussion by francisliardet** [rbaron/b-parasite#217] [Q&A] [answered]
Has anyone run into an issue where they can't set the transmit power in the project config due to "Missing dependencies" in VSCode?

I'm running nRF SDK/Toolchain v2.5.0, although I had the same issue when running @ v2.7.0 and v2.3.0.

I've linked screenshots of the compile warnings and my current build config.

Also I'm using [this module](https://www.cdebyte.com/products/E73-2G4M08S1C/), if that is relevant at all.

Any help would be appreciated, thanks!

![image](https://github.com/user-attachments/assets/6678e4ac-2869-42a1-8062-6643e1fa287b)
![image](https://github.com/user-attachments/assets/669b97fb-94c0-4d15-92ee-76705a0bc03a)

**Comment by rbaron** [comment]
Hey @francisliardet. Unfortunately I cannot reproduce the issue. I just double checked and we do have some recent clean builds on `2.5.x` ([example](https://github.com/rbaron/b-parasite/actions/runs/10734046697)). As a side note, I started tweaking the project to compile cleanly with the `2.7.x` (work in progress https://github.com/rbaron/b-parasite/pull/218).

With the `2.5.x` SDK + toolchain, can you reproduce the issue on a pristine build? Or equivalently, after removing all the `build*` directories in `ble/`?

**Reply by francisliardet** [reply]
Heya, thanks for the reply!

So I've done a few things and still haven't managed to quash that warning unfortunately:

- I deleted the build folder and made a pristine build on my current workspace
- I updated to SDK version 2.5.3 (from 2.5.0)
- I pulled a fresh copy of the 2.5 branch from github into a new workspace
- I pulled a fresh copy of the 2.7 branch from github into a new workspace and changed toolchain / SDK to 2.7.0

On that last point I managed to get rid of the `CONFIG_USE_SEGGER_RTT` warning, but the TX one is still there, pic attached below.

I think I'll set up a new, clean workspace on my laptop and try again - I'm hoping I've just busted something locally through all of the version changes and SDK updates.

If I were to start fresh, what version of Toolchain + SDK + b-parasite repo would be the most reliable?


![image](https://github.com/user-attachments/assets/aabb9972-0102-4d91-9378-ac37c7224024)

**Reply by rbaron** [reply]
Tks for the update. I think for now the 2.5 SDK/toolchain is the way to go. 

I noticed from the screenshot that you're on windows. I wonder if there's something odd going on with the extension there. Could you paste the full log from a 2.5 pristine build?

If you have time, could you try building it with docker instead?   I'd be curious to see if the error goes away. 

I hope I can finish the port to 2.7 when I'm back home in a week, but until then the 2.5 is the latest, most battle tested one.

**Reply by francisliardet** [reply, marked as answer]
So, it looks like it may have been a false alarm all along...

I managed to build everything using Docker successfully (using 2.5 SDK) and the functional range didn't change at all, even after what I had assumed was a 8db boost.

I did some tests building the firmware in both Docker and VSCode, changing the transmit power in the config each time, switching between +8db, +0db (no config), and -8db. 

As you can see, both Docker and VSCode gave the exact same result as measured via the "Signal Strength" HA entity, even though VSCode was consistently giving me warnings that my configuration wasn't making it through to the build.

Given these results, I'm pretty confident in assuming those warnings are total BS, and everything is totally fine, so I'll just ignore those messages for now and pretend everything is working normally!


![signal_strength](https://github.com/user-attachments/assets/b9170e47-5ac5-4b41-95da-93e936e3016a)

**Reply by rbaron** [reply]
Nice, I'm glad to hear. That's a nice visualization too. 

Happy hacking!
